src/core/hooks/use-field/use-field.ts

```diff
@@ -38,39 +44,103 @@ export const useField = <T, Err>(
   fieldDescriptor: GenericFieldDescriptor<T, Err>,
   controller?: FormController
 ): FieldHandle<T, Err> => {
-  const { methods } = useFormtsContext<object, Err>(controller);
+  const { methods, state } = useFormtsContext<object, Err>(controller);
+
+  const fieldState = useMemo(() => createFieldState(state, fieldDescriptor), [
+    state,
+    impl(fieldDescriptor).__path,
+  ]);
+
+  useSubscription(fieldState);
 
-  return createFieldHandle(fieldDescriptor, methods);
+  return createFieldHandle(fieldDescriptor, methods, fieldState, state);
+};
+
+type FieldState<T, Err> = Atom.Readonly<{
+  value: T;
+  touched: TouchedValues<T>;
+  error: Err | null;
+  isValid: boolean;
+  isValidating: boolean;
+}>;
+
+const createFieldState = <T, Err>(
+  state: FormtsAtomState<object, Err>,
+  field: FieldDescriptor<T, Err>,
+  includeChildrenDependencies: boolean = false
```

Comment:
maybe instead of function input (which will complicate atom cache feature) it can be derived from field type?
```ts
const includeChildrenDependencies = impl(field).__decoder.type === "object"
```

Reply:
my intention was not to base on field type but on hierarchy
imho children dependencies should be triggered only in top-level parent, initiator of `useField`, who will subscribe once to all branch changes